### PR TITLE
fix(massmailing): #MA-1182 display MAIL template with absence category

### PR DESCRIPTION
**Pull request by flmartineau**
## Describe your changes

- fixed display of massmailing MAIL templates in Viescolaire settings page for Absence category

## Checklist tests

- initialize Présences
- check if the template "Quotidien MAIL" has the Absence Category

## Issue ticket number and link

https://jira.support-ent.fr/browse/MA-1182

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: massmailing/src/main/java/fr/openent/massmailing/model/Mailing/Template.java [hunk 1]
```diff
@@ -1,102 +1,136 @@
...
     private String structureId;
     private String name;
     private String content;
     private MailingType type;
-    private String created;
-    private String owner;
-    private String category;
+        private String created;
+        private String owner;
+        private String category;
```
Comment (Gregprod33) [inline massmailing/src/main/java/fr/openent/massmailing/model/Mailing/Template.java]: Indentation

File: massmailing/src/main/java/fr/openent/massmailing/model/Mailing/Template.java [hunk 2]
```diff
@@ -1,102 +1,136 @@
...
+                String category) {
+            this.id = id;
+            this.structureId = structureId;
+            this.name = name;
+            this.content = content;
+            this.type = type;
+            this.created = created;
+            this.owner = owner;
+            this.category = category;
+        }
```
Comment (Gregprod33) [inline massmailing/src/main/java/fr/openent/massmailing/model/Mailing/Template.java]: indentation

File: massmailing/src/main/java/fr/openent/massmailing/service/impl/DefaultSettingsService.java [hunk 3]
```diff
@@ -33,7 +36,11 @@ public void getTemplates(MailingType type, String structure, List<String> listCa
         }
         queryBuilder.append(";");
 
-        Sql.getInstance().prepared(queryBuilder.toString(), params, SqlResult.validResultHandler(handler));
+
+        Sql.getInstance().prepared(queryBuilder.toString(), params,
+                SqlResult.validResultHandler(IModelHelper.sqlResultToIModel(promise, Template.class)));
```
Comment (Gregprod33) [inline massmailing/src/main/java/fr/openent/massmailing/service/impl/DefaultSettingsService.java]: Utilisation de la signature avec le message d'erreur :
SqlResult.validResultHandler(IModelHelper.sqlResultToIModel(promise, Template.class, errorMessage)

File: massmailing/src/main/java/fr/openent/massmailing/model/Mailing/Template.java [hunk 4]
```diff
@@ -1,102 +1,136 @@
...
 
     public Template(int id) {
-        this.id = id;
-    }
+            this.id = id;
+        }
```
Comment (Gregprod33) [inline massmailing/src/main/java/fr/openent/massmailing/model/Mailing/Template.java]: indentation

File: massmailing/src/main/java/fr/openent/massmailing/model/Mailing/Template.java [hunk 5]
```diff
@@ -1,102 +1,136 @@
...
 
-    public Template setOwner(String owner) {
-        this.owner = owner;
-        return this;
-    }
+        public Template setOwner(String owner) {
+            this.owner = owner;
+            return this;
+        }
```
Comment (Gregprod33) [inline massmailing/src/main/java/fr/openent/massmailing/model/Mailing/Template.java]: indentations

File: massmailing/src/main/java/fr/openent/massmailing/controller/SettingsController.java [hunk 6]
```diff
@@ -50,7 +51,9 @@ public void getAllTemplates(HttpServerRequest request) {
         }
 
         MailingType type = MailingType.valueOf(mailingType);
-        settingsService.getTemplates(type, structure, listCategory, arrayResponseHandler(request));
+        settingsService.getTemplates(type, structure, listCategory)
+                .onFailure(fail -> badRequest(request, fail.getMessage()))
```
Comment (Gregprod33) [inline massmailing/src/main/java/fr/openent/massmailing/controller/SettingsController.java]: .onFailure(fail -> renderError(request))

**Review by Gregprod33 — CHANGES_REQUESTED** [review]

File: massmailing/src/main/java/fr/openent/massmailing/model/Mailing/Template.java [hunk 7]
```diff
@@ -92,11 +108,29 @@ public Template setOwner(String owner) {
...
     }
 
     public Template setCategory(String category) {
         this.category = category;
         return this;
     }
+
+    @Override
+    public JsonObject toJson() {
+        return new JsonObject()
```
Comment (BastienBenard) [inline massmailing/src/main/java/fr/openent/massmailing/model/Mailing/Template.java:121]: IModelHelper.toJson stv

**Review by Sofianebernoussi — APPROVED** [review]

File: massmailing/src/main/java/fr/openent/massmailing/model/Mailing/Template.java [hunk 8]
```diff
@@ -92,11 +108,29 @@ public Template setOwner(String owner) {
     }
 
     public String getCategory() {
-        return category;
+        return (Objects.equals(category, "ABSENCE")) ? "ABSENCES" : category;
```
Comment (BastienBenard) [inline massmailing/src/main/java/fr/openent/massmailing/model/Mailing/Template.java]: Magic string

File: massmailing/src/main/java/fr/openent/massmailing/model/Mailing/Template.java (hunk 7 above)
Comment (flmartineau) [reply massmailing/src/main/java/fr/openent/massmailing/model/Mailing/Template.java:121]: Ca ne fonctionnerait pas car je veux qu'il utilise la méthode getCategory qui n'est pas appellée sinon.

**Pull request merged**